### PR TITLE
Documentation about the new watcher

### PR DESCRIPTION
**Pull request by nono**

### CODE REVIEW

**Comment by nono** [comment]
[Ajouter de la documentation sur le nouveau watcher](https://trello.com/c/aT6PDiaq/1373-ajouter-de-la-documentation-sur-le-nouveau-watcher)

File: doc/developer/design.md [hunk 1]
```diff
@@ -168,4 +168,30 @@ The permissions are different on Unix and on Windows. So, the `executable`
 field is just ignored on Windows.
 
 
+Local watcher
+-------------
+
+Historically, the local watcher was based on the chokidar library. Chokidar is
+an high-level library that tries to mask the differences between the API for
```
Comment (enguerran) [inline doc/developer/design.md]: s/an/a/

File: doc/developer/design.md [hunk 2]
```diff
@@ -168,4 +168,30 @@ The permissions are different on Unix and on Windows. So, the `executable`
...
+watching files on 3 OSes: ReadDirectoryChangesW on Windows, FsEvents on macOS,
+and inotify on Linux. But, we discover that doing so leads to some pitfalls,
+for our use case. In particular, on Windows, Chokidar works with polling or
```
Comment (enguerran) [inline doc/developer/design.md]: s/our use case/our use cases/

Comment (enguerran) [reply doc/developer/design.md]: But not sure ;-)

File: doc/developer/design.md [hunk 3]
```diff
@@ -168,4 +168,30 @@ The permissions are different on Unix and on Windows. So, the `executable`
...
+with creating a new watcher per direcotry, while a single watcher on the sync
```
Comment (enguerran) [inline doc/developer/design.md]: s/direcotry/directory/

File: doc/developer/design.md [hunk 4]
```diff
@@ -168,4 +168,30 @@ The permissions are different on Unix and on Windows. So, the `executable`
 field is just ignored on Windows.
 
 
+Local watcher
+-------------
+
+Historically, the local watcher was based on the chokidar library. Chokidar is
+a high-level library that tries to mask the differences between the API for
+watching files on 3 OSes: ReadDirectoryChangesW on Windows, FsEvents on macOS,
+and inotify on Linux. But, we discover that doing so leads to some pitfalls,
```
Comment (taratatach) [inline doc/developer/design.md]: `s/discover/discovered`

File: doc/developer/design.md [hunk 5]
```diff
@@ -168,4 +168,30 @@ The permissions are different on Unix and on Windows. So, the `executable`
 field is just ignored on Windows.
 
 
+Local watcher
+-------------
+
+Historically, the local watcher was based on the chokidar library. Chokidar is
+a high-level library that tries to mask the differences between the API for
```
Comment (taratatach) [inline doc/developer/design.md]: `s/API/APIs`

File: doc/developer/design.md (hunk 4 above)
Comment (taratatach) [inline doc/developer/design.md]: `s/leads to/has`

File: doc/developer/design.md [hunk 6]
```diff
@@ -168,4 +168,30 @@ The permissions are different on Unix and on Windows. So, the `executable`
...
+watching files on 3 OSes: ReadDirectoryChangesW on Windows, FsEvents on macOS,
+and inotify on Linux. But, we discover that doing so leads to some pitfalls,
+for our use cases. In particular, on Windows, Chokidar works with polling or
+with creating a new watcher per directory, while a single watcher on the sync
```
Comment (taratatach) [inline doc/developer/design.md]: `s/with creating/by creating`

Comment (enguerran) [reply doc/developer/design.md]: "with the creation" or "with creating" or "by creating" are all correct, according to deepl:

https://www.deepl.com/translator#fr/en/En%20particulier%2C%20sous%20Windows%2C%20Chokidar%20travaille%20avec%20polling%20ou%20avec%20la%20cr%C3%A9ation%20d'un%20nouveau%20watcher%20par%20r%C3%A9pertoire

**Pull request merged**